src/test/java/com/google/step/datamanager/DealTagManagerDatastoreTest.java

```diff
@@ -130,4 +132,16 @@ public void getDealsWithTag_tagDoesNotExist() {
 
     assertTrue(dealIds.isEmpty());
   }
+
+  @Test
+  public void testDeleteAllTagsOfDeal_otherDealNotAffected() {
+    dealTagManagerDatastore.updateTagsOfDeal(DEAL_ID_A, Arrays.asList(TAG_ID_A));
+    dealTagManagerDatastore.updateTagsOfDeal(DEAL_ID_B, Arrays.asList(TAG_ID_A));
+    dealTagManagerDatastore.deleteAllTagsOfDeal(DEAL_ID_B);
+
+    List<Long> tagsA = dealTagManagerDatastore.getTagIdsOfDeal(DEAL_ID_A);
+    List<Long> tagsB = dealTagManagerDatastore.getTagIdsOfDeal(DEAL_ID_B);
+    assertEquals(Arrays.asList(TAG_ID_A), tagsA);
+    assertEquals(new ArrayList<>(), tagsB);
```

Comment:
I think a better way of checking it may be to assert that the length of tagsB == 0 instead of checking it's a new ArrayList instance.